src/common/util.js

```diff
@@ -29,3 +29,9 @@ export const arrayRotate = (arr, count) => {
   arr.push.apply(arr, arr.splice(0, count))
   return arr
 }
+
+// set to true to turn on console logging
+const debug = false
+export const log = (message) => {
+  if (debug) { console.log(message) }
```

Comment:
If this is always false, does this just get "optimized out"? Nifty feature, if it did.

Reply:
I see that some of the other information, specifically the file/line are set to this log location, instead of the source file.

Does it make more sense to do something like `console.debug()`? I am not familiar with the nuances here. I wouldn't want it to be slower for the end user, but I also don't mind if someone reads our debug logs at sandify.org.

Reply:
I don't know how much slower spitting out log messages would make pattern rendering, but on principle I don't see a reason to spam the logs unnecessary, especially if we're just using for debugging purposes when something isn't working. Reading up on console.debug, it seems to be the same as console.log, just hidden behind a filter. As far as line number goes, the log message itself tells you where you are. Anyway, my inclination is to leave it as-is. We can always use console.log (or debug) in places where things go wrong, like you have done in boundary condition logic.